### ISSUE TITLE
Successful end-to-end login and connection close

### ISSUE THREAD

**Issue by chrisdone** [NewMountain/odbcJsonGateway#1]
Implement successful login end-to-end: a service capable of accepting
connections and closing them cleanly.

- [ ] Setup our freetds client in the repo.
- [x] Create a server capable of accepting connections with the right
  login.
- [x] Make a test suite that runs a temp server and connects to it
  with freetds. Test successful and insuccessful login.

This consists of these requests/responses

* Pre-Login Request         
* Login Request             
* Login Response

**Comment by chrisdone** [comment]
First successful PRELOGIN handshake

<img width="1440" alt="screenshot 2017-10-23 18 25 36" src="https://user-images.githubusercontent.com/11019/31903379-d99f2514-b81f-11e7-8622-d4c1d4c7c2bd.png">

That's 12=PRELOGIN, 04=PRELOGIN REPLY, 10=LOGIN ... next step is to handle LOGIN. :+1:

**Comment by chrisdone** [comment]
Login process complete. 

Works with tsql, sqlcmd and ODBC MS SQL, and also Tableau. I've implemented a test suite that runs the service and uses ODBC MS SQL to login to it.

Documentation here: https://github.com/NewMountain/odbcJsonGateway/blob/master/tds/README.md

Closing. Next step is handling data queries.

---  

This took a while; every client behaves differently. There is apparently a timing issue in which MS's ODBC driver will give up if it doesn't get login handshake replies immediately. Any delay and it kills the connection without remorse. More gory details on request.
